### PR TITLE
Raycast

### PR DESCRIPTION
**Pull request by RobotCaleb**
This also has the beginnings of the remove entity tool tweaks. Just renames so far. I don't want to decouple, though.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -61,7 +61,7 @@ set(EN_SOURCES
     src/tools/fire_projectile.cc
     src/tools/remove_block.cc
     src/tools/remove_surface.cc
-    src/tools/remove_surface_entity.cc
+    src/tools/remove_entity.cc
```
Comment (chrisforbes) [inline CMakeLists.txt]: will need a matching VS change

Comment (RobotCaleb) [reply CMakeLists.txt]: Done

File: main.cc [hunk 2]
```diff
@@ -727,10 +725,18 @@ struct play_state : game_state {
...
-            /* both tool use and overlays need the raycast itself */
-            block_raycast_info rc;
-            ship->raycast(pl.eye, pl.dir, MAX_REACH_DISTANCE, &rc);
-            raycast_dist = rc.t;
+            /* both tool use and overlays need the raycast_block itself */
+            raycast_info rc;
+            ship->raycast_block(pl.eye, pl.dir, MAX_REACH_DISTANCE, &rc.block);
+
+            /* interact with ents. do this /after/
+             * anything that may delete the entity
```
Comment (chrisforbes) [inline main.cc]: This comment makes no sense now.

Comment (RobotCaleb) [reply main.cc]: yep

Comment (RobotCaleb) [reply main.cc]: fixt

File: src/char.cc [hunk 3]
```diff
@@ -859,30 +860,35 @@ void en_char_controller::stand(btCollisionWorld *collisionWorld)
...
     btVector3 end_bt = glm_to_bt(end);
     en_ray_result_callback callback(ignore, start_bt, end_bt);
     world->rayTest(start_bt, end_bt, callback);
 
     if (callback.hasHit()) {
-        return (phys_ent_ref *)callback.m_collisionObject->getUserPointer();
+        if (callback.m_collisionObject->getUserPointer()) {
+            rc->hit = true;
+            rc->entity = ((phys_ent_ref *) callback.m_collisionObject->getUserPointer())->ce;
+            rc->hitCoord = glm::vec3(callback.m_hitPointWorld.x(), callback.m_hitPointWorld.y(),
```
Comment (chrisforbes) [inline src/char.cc]: this is what bt_to_glm does.

Comment (RobotCaleb) [reply src/char.cc]: Thanks! I was briefly looking for.

Comment (chrisforbes) [inline src/char.cc]: make this capture the normal as well, and throw away generic

Comment (chrisforbes) [reply src/char.cc]: (hitting the world is rc->hit = true, rc->entity = invalid)

Comment (RobotCaleb) [reply src/char.cc]: alright. I'll add c_entity::is_valid() as well

Comment (RobotCaleb) [reply src/char.cc]: fixt

Comment (RobotCaleb) [reply src/char.cc]: fixt

**Pull request merged**